### PR TITLE
Added merge crate

### LINKED ISSUE #21: Use the merge crate
Use the [merge](https://docs.rs/merge/latest/merge/) crate instead of `Config.merge()` and `Format.merge()`.
This would simplify the code and make adding new options easier by introducing a new dependency.
I could implement this if i get the green light.

### PR DESCRIPTION
**Pull request by OliverKovacs**
Sadly i couldn't entirely get rid of the custom merge function as the crate doesn't support nested structs.

Closes #21

### CODE REVIEW

File: src/config.rs [hunk 1]
```diff
@@ -28,37 +29,20 @@ pub struct Format {
...
-    fn merge(self, other: Config) -> Self {
-        Self {
-            id: self.id.or(other.id),
-            hosts: self.hosts.or(other.hosts),
-            format: match self.format {
-                Some(format) => Some(format.merge(other.format.unwrap())),
-                _ => other.format
-            }
-        }
+    fn merge_custom(mut self, other: Config) -> Self {
```
Comment (JakeStanger) [inline src/config.rs:40]: Is there a reason this was renamed to `merge_custom`?

**Comment by JakeStanger** [comment]
Thanks for doing this!

> the crate doesn't support nested structs.

I'' have a look at the upstream package and see if we can get support for that added as soon as I get time. I'll hold off merging this until then.

File: src/config.rs (hunk 1 above)
Comment (OliverKovacs) [reply src/config.rs:40]: `derive(Merge)` creates `merge` so I needed another name.

**Comment by JakeStanger** [comment]
Adding nested struct support is going to be harder than it's worth. Just had a proper look at this code and it's absolutely fine as-is, so happy to merge. Thanks again!

**Pull request merged**